server/src/utils/Utils.ts

```diff
@@ -39,5 +39,94 @@ export const selectRandomWeighted = (items, weights, n) => {
     choices.push(newItem);
   }
 
-  return choices;
+  // For randomly weighted mutations
+  if (n==1){
```

Comment:
The case where `n == 1` is already handled by the above logic, so we shouldn't need to handle it separately.
I think it's better if we always return an array for sake of consistency, even if `n == 1`